### PR TITLE
Added MongoDB Driver and a MongoDBCollection

### PR DESCRIPTION
**Pull request by tigerstrikemedia**
The function Collection::summatory() is not there, and would fail badly if called, as the function group() doesn't exist on the new driver Collection class.

Its a rough start, but is working for all the basics on my machine, Phalcon Version: 2.1.0 RC 1 on PHP version: 7.0.7 looking for some feedback and requirements on moving forwards.

Ben

### CODE REVIEW

**Comment by sergeyklay** [comment]
@tigerstrikemedia Thank you for contributing! Could you please fix PSR errors?

**Comment by sergeyklay** [comment]
> Line exceeds 120 characters

This is minor changes

> Method name "StreamWrapper::stream_read" is not in camel caps format

Are you absolutely sure we need `StreamWrapper::stream_read`, not `StreamWrapper::streamRead`? Could you please explain why?

**Comment by tigerstrikemedia** [comment]
The stream functions are a stream wrapper for GridFS functionality, and follow the [PHP functions](http://php.net/manual/en/book.stream.php). Really not knowing enough about gridFS i'm not sure if renaming them is a good move or not. 

I will rename the functions in the Functions class as they all seem to be helpers for the PHP functionality.

**Comment by tigerstrikemedia** [comment]
@sergeyklay 

Ok, we are down to the stream_XX functions and some warnings about the _ at the start of the variable / function names to indicate visibility.

**Comment by sergeyklay** [comment]
@tigerstrikemedia Well, remains two classes for PSR fixes

**Comment by tigerstrikemedia** [comment]
@sergeyklay 

`$_disableEvents` `_getResultset()` and `_exists` were taken verbatim from the collection.zep file used to build the Collection class in phalcon core. Should I change them here?

`stream_xx ` functions are not able to be changed as doing so would mean that GridFS is unusable in phalcon.

Having read up on stream wrappers, they are an unofficial interface requirement of  stream wrappers, but are not implemented as an interface for [good reason](http://php.net/manual/en/class.streamwrapper.php#100549)

**Comment by sergeyklay** [comment]
@tigerstrikemedia Okay. RIght now just use this http://stackoverflow.com/questions/16039362/how-can-i-suppress-phpcs-warnings-using-comments

File: Library/Phalcon/Mvc/MongoCollection.php [hunk 1]
```diff
@@ -174,17 +176,18 @@ public static function findFirst(array $parameters = null)
...
      *
      * @return array
+     * @throws Exception
      */
+    // @codingStandardsIgnoreStart
     protected static function _getResultset($params, \Phalcon\Mvc\CollectionInterface $collection, $connection, $unique)
     {
+        // @codingStandardsIgnoreEnd
         //		var source, mongoCollection, conditions, base, documentsCursor,
         //			fields, skip, limit, sort, document, collections, className;
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/MongoCollection.php]: > var source, mongoCollection, conditions, base, documentsCursor,

what is?

File: Library/Phalcon/Mvc/MongoCollection.php [hunk 2]
```diff
@@ -393,8 +396,10 @@ public function delete()
      *
      * @return boolean
      */
+    // @codingStandardsIgnoreStart
     protected function _exists($collection)
     {
+        // @codingStandardsIgnoreStart
         //var id, mongoId;
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/MongoCollection.php]: > //var id, mongoId; 

?

**Comment by sergeyklay** [comment]
Well, I like it. Fine! Could you please clean php from zephir? :)

**Comment by sergeyklay** [comment]
Thanks

**Comment by sergeyklay** [comment]
@tigerstrikemedia Could you please update `README.md`?

**Pull request merged**